### PR TITLE
Update to simplify aggregates, events and repositories scenario sample

### PR DESCRIPTION
**Pull request by mysticmind**

### CODE REVIEW

File: src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs [hunk 1]
```diff
@@ -178,39 +203,29 @@ public abstract class AggregateBase
...
+            _uncommittedEvents.Clear();
         }
 
         protected void RaiseEvent(object @event)
         {
-            ApplyEvent(@event);
-            uncommittedEvents.Add(@event);
-        }
+            // using ReflectionMagic library to invoke the Apply method
+            this.AsDynamic().Apply(@event);
```
[the remark this answers is not in the document]
Comment (mysticmind) [reply src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs]: There is an easier fix, will update shortly.

[the remark this answers is not in the document]
Comment (mysticmind) [reply src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs]: @oskardudycz you can take a took at the changes now, removed usage of RelfectionMagic library and the calls are very explicit and sampler to understand.

**Comment by mysticmind** [comment]
@oskardudycz @jokokko Please note that I am only trying to get a good middle path and simpler sample, based on both of your changes.

File: src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs [hunk 2]
```diff
@@ -70,6 +73,23 @@ public void CanGuardVersion()
...
+            var invoice = CreateInvoice();
+            invoice.Version.ShouldBe(3);
+            repository.Store(invoice);
+
+            // assert version was incremented properly
+            var invoiceFromRepository = repository.Load<Invoice>(invoice.Id);
+            invoiceFromRepository.Version.ShouldBe(3);
+
+            // update aggregate
+            invoiceFromRepository.AddLine(100, 23, "Some nice product with 23% VAT");
```
Comment (oskardudycz) [inline src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs]: Those scenarios are missing:
```
repository.Store(invoiceFromRepository);

            // assert version was incremented properly
            invoiceFromRepository = repository.Load<Invoice>(invoice.Id);
            invoiceFromRepository.Version.ShouldBe(4);
```
from my PR https://github.com/JasperFx/marten/pull/1299/files#diff-22165075f6da4b37fe6d31e83b98d376R91

File: src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs [hunk 3]
```diff
@@ -178,39 +213,26 @@ public abstract class AggregateBase
...
-        // Infrastructure for raising events & registering handlers
-
-        protected void Register<T>(Action<T> handle)
-        {
-            handlers[typeof(T)] = e => handle((T)e);
+            _uncommittedEvents.Clear();
         }
 
-        protected void RaiseEvent(object @event)
+        protected void AddUncommittedEvents(object @event)
```
Comment (oskardudycz) [inline src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs]:
```suggestion
        protected void AddUncommittedEvent(object @event)
```

File: src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs (hunk 2 above)
Comment (mysticmind) [reply src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs]: done, added it

File: src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs (hunk 3 above)
Comment (mysticmind) [reply src/Marten.Testing/Events/ScenarioAggregateAndRepository.cs]: renamed the method

**Pull request merged**